### ISSUE TITLE
RTX 5060 Ti (GB206, sm_120): Xid 175 GSP RPC timeout (DUP_OBJECT) on CUDA context init — headless secondary GPU

### ISSUE THREAD

**Issue by basiphobe** [NVIDIA/open-gpu-kernel-modules#1242]
### NVIDIA Open GPU Kernel Modules Version

595.71.05 (Ubuntu package nvidia-driver-595-open 595.71.05-0ubuntu0.26.04.1, srcversion 58D233B8E3F4A2973D73151) 

### Please confirm this issue does not happen with the proprietary driver (of the same version). This issue tracker is only for bugs specific to the open kernel driver.

- [ ] I confirm that this does not happen with the proprietary driver package.

### Operating System and Version

Ubuntu 26.04 LTS

### Kernel Release

 Linux aaron-pc 7.0.0-27-generic #27-Ubuntu SMP PREEMPT_DYNAMIC Thu Jun 18 19:13:49 UTC 2026 x86_64 

### Please confirm you are running a stable release kernel (e.g. not a -rc). We do not accept bug reports for unreleased kernels.

- [x] I am running on a stable kernel release.

### Hardware: GPU

GPU 0: NVIDIA GeForce RTX 4070 SUPER (UUID: GPU-a343e71b-2463-27f5-3191-c3f86e6404e4) GPU 1: NVIDIA GeForce RTX 5060 Ti (UUID: GPU-fe00ae81-39f7-3d58-7df7-4732507d377c)

### Describe the bug

Any attempt to initialize a CUDA context on the RTX 5060 Ti (GPU1, headless/no display) results in Xid 175 — GSP RPC timeout waiting for DUP_OBJECT response — after which the GPU requires reset and all CUDA calls return error 46 ("device busy or unavailable"). This also poisons nvidia_uvm globally, breaking CUDA on the co-installed RTX 4070 SUPER. Full system reboot required to recover.

Kernel log:
NVRM: Xid (PCI:0000:16:00): 175, pid=22376, name=llama-server, Timeout after 10s of waiting for RPC response from GPU1 GSP! Expected function 21 (DUP_OBJECT) sequence 3200 (0x0 0x0).
NVRM: Xid (PCI:0000:16:00): 154, GPU recovery action changed from 0x0 (None) to 0x1 (GPU Reset Required)

Note: PCIe ASPM must be disabled (pcie_aspm=off kernel param) and nvidia-persistenced enabled to reach this point — without those, an earlier Xid 32 / UVM fatal 0x60 fires during nvidia_uvm module init instead. Both symptoms trace to the same GPU (5060 Ti, PCI 0000:16:00.0).

The 5060 Ti is running at PCIe x4 (slot limitation): 63.012 Gb/s available, limited by 16.0 GT/s x4 link at 0000:00:02.2.

### To Reproduce

1. Install RTX 5060 Ti as a secondary headless GPU (no display connected) alongside an RTX 4070 SUPER
2. Install nvidia-driver-595-open 595.71.05
3. Add pcie_aspm=off to GRUB_CMDLINE_LINUX_DEFAULT and run update-grub
4. Enable nvidia-persistenced (remove --no-persistence-mode flag from service)
5. Reboot
6. Run any CUDA application targeting the 5060 Ti, e.g.:
   python3 -c "import torch; torch.cuda.set_device(1); torch.cuda.memory_allocated()"
   or: CUDA_VISIBLE_DEVICES=1 ollama serve
7. Within 10 seconds: Xid 175 fires, GPU requires reset, CUDA broken on both GPUs until reboot

### Bug Incidence

Always

### nvidia-bug-report.log.gz

[nvidia-bug-report.log.gz](https://github.com/user-attachments/files/30060000/nvidia-bug-report.log.gz)

**Comment by basiphobe-personal** [comment]
Still reproducible on 595.84-0ubuntu0.26.04.1, kernel 7.0.0-28-generic (Ubuntu 26.04 LTS).

Xid 175 (DUP_OBJECT, fn 21, sequence 3200) fires within 10 seconds of any CUDA context init on the 5060 Ti. No change in behavior from 595.71.05. No NVIDIA response to this issue or related issues #1230 or #1111 as of 2026-07-31.

**Comment by Waleed794** [comment]
### Additional Data Point: RTX 5070 Ti (GB203 / Blackwell) eGPU over USB4

I am hitting a very similar **Xid 175 GSP RPC timeout** on a Blackwell secondary GPU using `nvidia-open` 595.84 on Ubuntu 26.04 LTS.

#### System Configuration
* **Host:** Dell Laptop (Intel iGPU + RTX 4050 Laptop dGPU)
* **Secondary GPU:** RTX 5070 Ti (GB203) via USB4 eGPU (`0000:06:00.0`, PCIe x4 link)
* **Driver:** `nvidia-driver-595-open` (595.84-0ubuntu0.26.04.1)
* **Kernel:** 6.x / 7.0-generic (Ubuntu 26.04)

#### Logs
When Mutter/gnome-shell or DRM mode-setting attempts to enumerate the secondary Blackwell eGPU alongside the internal GPUs, the driver hangs for 10 seconds before throwing Xid 175:

```text
NVRM: GPU1 nvCheckOkFailedNoLog: Check failed: Generic Error: Invalid state [NV_ERR_INVALID_STATE] (0x00000040) returned from pRmApi->Control @ gpu.c:7450
NVRM: Xid (PCI:0000:06:00): 175, pid=5948, name=gnome-shell, Timeout after 10s of waiting for RPC response from GPU1 GSP! Expected function 76 (GSP_RM_CONTROL) sequence 1207 (0x20800a9a 0x8).
NVRM: Xid (PCI:0000:06:00): 154, GPU recovery action changed from 0x0 (None) to 0x1 (GPU Reset Required)
```

#### Partial Workaround Observed
Masking the card from GNOME/Mutter via `TAG+="mutter-device-ignore"` stops gnome-shell from issuing the DRM mode-setting queries, preventing Xid 175 on boot. Once masked from display management, CUDA and nvidia-smi function over the UVM/character device layer.

However, any attempt to let a display manager probe or manage the GB203 secondary GPU over the PCIe x4/USB4 topology triggers the 10-second GSP RPC stall on `function 76 (GSP_RM_CONTROL)`.

**Comment by basiphobe-personal** [comment]
Thanks for the additional data point, @Waleed794. Interesting that we're hitting the same Xid 175 / 10-second GSP RPC timeout pattern on different Blackwell SKUs (GB206 vs GB203) and different trigger paths.

A few observations that may help narrow this down for NVIDIA:

• My trigger:  DUP_OBJECT  (fn 21, sequence 3200) — fires during CUDA context initialization, headless GPU, no display manager involved
• Your trigger:  GSP_RM_CONTROL  (fn 76, sequence 1207) — fires when gnome-shell/Mutter attempts display mode-setting enumeration

So these are two distinct GSP RPC call sites failing, suggesting the issue isn't specific to one code path but rather something more fundamental about how secondary Blackwell GPUs handle GSP RPC calls — possibly related to the PCIe x4 link both our setups share.

Your workaround (masking from Mutter via  TAG+="mutter-device-ignore" ) is useful context — it implies the display-triggered variant can be avoided by preventing DRM enumeration. Unfortunately my failure mode is pure CUDA init with no display manager in the loop, so there's no equivalent avoidance strategy on my end.

Both cases result in Xid 154 (GPU Reset Required) and nvidia_uvm poisoning that requires a full reboot to clear.
